processor/tailsamplingprocessor/config.go

```diff
@@ -66,8 +66,15 @@ type NumericAttributeCfg struct {
 type StringAttributeCfg struct {
 	// Tag that the filter is going to be matching against.
 	Key string `mapstructure:"key"`
-	// Values is the set of attribute values that if any is equal to the actual attribute value to be considered a match.
+	// Values indicate the set of values or regular expressions to use when matching against attribute values.
+	// StringAttribute Policy will apply exact value match on Values unless RegexMatchEnabled is true.
 	Values []string `mapstructure:"values"`
+	// RegexMatchEnabled determines whether match attribute values by regexp string.
+	RegexMatchEnabled bool `mapstructure:"enabled_regex_matching"`
+	// CacheSize is the maximum number of attribute entries of LRU Cache that stores the matched result
+	// from the regular expressions defined in Values.
+	// CacheSize will not be used if RegexMatchEnabled is set to false.
+	CacheSize int `mapstructure:"cache_max_size"`
```

Comment:
Use `CacheMaxSize` or change the struct tag to `cache_size`, better keep them consistent (same for `RegexMatchEnabled`). Though only the struct tag is the actual 'public' API.

Reply:
Updated